rudder-core/src/main/scala/com/normation/rudder/reports/execution/ReportsExecutionService.scala

```diff
@@ -178,6 +181,27 @@ class ReportsExecutionService (
       case scala.util.Failure(ex) =>
         logger.error("An error occured when trying to update the cache of last changes", ex)
     }
+
+    future {
+      for {
+        updated <- cachedCompliance.checkAndUpdateCache(updatedNodeIds)
```

Comment:
For point 1/ & 2/, see update coming. 

For 3, no because we need to know anything about new runs, not only changes - so we really want all new (finished) runs